sympy/utilities/tests/test_autowrap.py

```diff
@@ -17,6 +17,35 @@
     CCodeGen, C99CodeGen, CodeGenArgumentListError, make_routine
 )
 
+class TmpFileManager:
+    """
+    A class to track record of every temporary files created by the tests.
+    """
+    tmp_files = set('')
+    tmp_folders = set('')
+
+    @classmethod
+    def tmp_file(cls, name=''):
+        cls.tmp_files.add(name)
+        return name
+
+    @classmethod
+    def tmp_folder(cls, name=''):
+        cls.tmp_folders.add(name)
+        return name
+
+    @classmethod
+    def cleanup(cls):
+        while cls.tmp_files:
+            file = cls.tmp_files.pop()
+            try:
+                os.remove(file)
+            except OSError:
+                # If the file doesn't exist, for instance, if the test failed.
+                pass
+        while cls.tmp_folders:
+            folder = cls.tmp_folders.pop()
+            shutil.rmtree(folder)
```

Comment:
Doesn't this delete the directory and everything in it? Can't you just use this and skip all the other code that tracks specific files?

Reply:
@moorepants 
Actually, some files get generated inside sympy directory. and I don't think `shutil.rmtree` works for a specific file.

Reply:
Then that's a different issue. We need to patch the codegen to fix that. This is just a bandaid on the actual problem if that is true.

Reply:
@moorepants 
I think it can simply be fixed by adding capability to specify directory for `.pyx` files in `_prepare_files()`.